client/internal/packagessyncer.go

```diff
@@ -280,6 +280,12 @@ func (s *packagesSyncer) downloadFile(ctx context.Context, pkgName string, file
 		return fmt.Errorf("cannot download file from %s: %v", file.DownloadUrl, err)
 	}
 
+	if file.Headers != nil && len(file.Headers.Headers) > 0 {
+		for _, h := range file.Headers.Headers {
+			req.Header.Add(h.GetKey(), h.GetValue())
```

Comment:
Using the `GetXXX` helpers that contain the nilness check to avoid doing it explicitly.